docs/content/envoy-introduction.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Overview & Architecture
+kind: envoy
+weight: 1
+---
+
+[Envoy](https://www.envoyproxy.io/docs/envoy/latest/intro/what_is_envoy) is a
+L7 proxy and communication bus designed for large modern service oriented
+architectures. Envoy (v1.7.0+) supports an [External Authorization
+filter](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html)
+which calls an authorization service to check if the incoming request is
+authorized or not.
+
+This feature makes it possible to delegate authorization decisions to an
+external service and also makes the request context available to the service. The request context contains information
+such as the source of a network activity, destination of a network activity, the network request (eg. http request).
+All this information can be used by the external service to make an informed decision about the fate of the
+incoming request received by Envoy.
+
+## What is OPA-Envoy Plugin?
+
+[OPA-Envoy](https://github.com/open-policy-agent/opa-envoy-plugin) plugin extends OPA with a gRPC server that
+implements the [Envoy External Authorization API](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html).
+You can use this version of OPA to enforce fine-grained, context-aware access control policies with Envoy _without_
+modifying your microservice.
+
+## How does it work?
+
+In addition to the Envoy sidecar, your application pods will include an OPA
+sidecar. When Envoy receives API requests destined for your
+microservice, it checks with OPA to decide if the request should be allowed.
+
+Evaluating policies locally with Envoy is preferable because it
+avoids introducing a network hop (which has implications on performance and
+availability) in order to perform the authorization check.
+
+{{< figure src="envoy-ext-authz-flow.png" width="150" caption="Envoy External Authorization Flow" >}}
+
+> 💡 The OPA-Envoy plugin can be deployed in a Kubernetes environment as a sidecar or it can also be co-located next to a
```

Comment:
Nit: I'd probably change this sentence to clarify that OPA-Envoy can be be used outside of Kubernetes. Something like:

```
> 💡 The OPA-Envoy plugin is frequently deployed in Kubernetes environments as a sidecar container however it can also be used in other environments as a standalone process running next to Envoy.
```

Reply:
Updated